presto-main/src/main/java/com/facebook/presto/cost/HistoricalPlanStatisticsUtil.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.cost;
+
+import com.facebook.presto.spi.statistics.HistoricalPlanStatistics;
+import com.facebook.presto.spi.statistics.PlanStatistics;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static java.lang.Double.isNaN;
+
+public class HistoricalPlanStatisticsUtil
+{
+    // Store historical data for last ``MAX_LAST_RUNS`` runs at maximum.
+    private static final int MAX_LAST_RUNS = 10;
```

Comment:
make these configurable. Maybe have a special HBO config prefix though or all the HBO configurations could get a bit unwieldy. I also wonder if these are the right metrics, or if instead we should have some size limit on historical runs, and remove any old runs that put you over the size limit.

Reply:
Simplified this to just store max last runs, it is directly equivalent to size